Kibana highlighting issue not fixed?

New 1.6.0 install with new indices, in Kibana discover searching for "execute":
```
[2015-06-11 13:18:59,718][DEBUG][action.search.type       ] [Laura Dean] [29] Failed to execute fetch phase
org.elasticsearch.search.fetch.FetchPhaseExecutionException: [logstash-2015.06.11][4]: query[filtered(_all:execute)->BooleanFilter(+cache(@timestamp:[1433938739612 TO 1434025139612]
))],from[0],size[500],sort[<custom:"@timestamp": org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource@31c250a5>!]: Fetch Failed [Failed to highlight field [m
essage.raw]]
        at org.elasticsearch.search.highlight.PlainHighlighter.highlight(PlainHighlighter.java:133)
        at org.elasticsearch.search.highlight.HighlightPhase.hitExecute(HighlightPhase.java:133)
        at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:194)
        at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:504)
        at org.elasticsearch.search.action.SearchServiceTransportAction$17.call(SearchServiceTransportAction.java:452)
        at org.elasticsearch.search.action.SearchServiceTransportAction$17.call(SearchServiceTransportAction.java:449)
        at org.elasticsearch.search.action.SearchServiceTransportAction$23.run(SearchServiceTransportAction.java:559)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.RuntimeException: org.apache.lucene.util.BytesRefHash$MaxBytesLengthExceededException: bytes can be at most 32766 in length; got 173493
        at org.apache.lucene.index.memory.MemoryIndex.addField(MemoryIndex.java:493)
        at org.apache.lucene.index.memory.MemoryIndex.addField(MemoryIndex.java:396)
        at org.apache.lucene.index.memory.MemoryIndex.addField(MemoryIndex.java:371)
        at org.apache.lucene.index.memory.MemoryIndex.addField(MemoryIndex.java:351)
        at org.apache.lucene.search.highlight.WeightedSpanTermExtractor.getLeafContext(WeightedSpanTermExtractor.java:360)
        at org.apache.lucene.search.highlight.WeightedSpanTermExtractor.extract(WeightedSpanTermExtractor.java:216)
        at org.apache.lucene.search.highlight.WeightedSpanTermExtractor.getWeightedSpanTerms(WeightedSpanTermExtractor.java:474)
        at org.apache.lucene.search.highlight.QueryScorer.initExtractor(QueryScorer.java:217)
        at org.apache.lucene.search.highlight.QueryScorer.init(QueryScorer.java:186)
        at org.apache.lucene.search.highlight.Highlighter.getBestTextFragments(Highlighter.java:197)
        at org.elasticsearch.search.highlight.PlainHighlighter.highlight(PlainHighlighter.java:119)
        ... 9 more
Caused by: org.apache.lucene.util.BytesRefHash$MaxBytesLengthExceededException: bytes can be at most 32766 in length; got 173493
        at org.apache.lucene.util.BytesRefHash.add(BytesRefHash.java:284)
        at org.apache.lucene.index.memory.MemoryIndex.addField(MemoryIndex.java:467)
        ... 19 more
```
I thought this was fixed in #9881 - but for some reason ES is still trying to highlight message.raw which is not_analyzed. The comment in the Kibana issue https://github.com/elastic/kibana/issues/2782#issuecomment-105224019 says that long terms shouldn't get into the index and not_analyzed should be skipped - but neither seem to be happening here.

Reproduction in this [gist](https://gist.github.com/jimmyjones2/0be6664a471a30b2e624) then add index to Kibana and in discover search for "execute". My original setup involves Elasticsearch sending its own logs to logstash via log4j socket, then ingesting into another elasticsearch instance - can include details here if needed.

--- a/src/main/java/org/elasticsearch/search/highlight/PlainHighlighter.java
+++ b/src/main/java/org/elasticsearch/search/highlight/PlainHighlighter.java
@@ -28,6 +28,7 @@ import org.apache.lucene.search.highlight.*;
 import org.apache.lucene.util.BytesRefHash;
 import org.apache.lucene.util.CollectionUtil;
 import org.elasticsearch.ElasticsearchIllegalArgumentException;
+import org.elasticsearch.ExceptionsHelper;
 import org.elasticsearch.common.text.StringText;
 import org.elasticsearch.common.text.Text;
 import org.elasticsearch.index.mapper.FieldMapper;
@@ -124,7 +125,7 @@ public class PlainHighlighter implements Highlighter {
                 }
             }
         } catch (Exception e) {
-            if (e instanceof BytesRefHash.MaxBytesLengthExceededException) {
+            if (ExceptionsHelper.unwrap(e, BytesRefHash.MaxBytesLengthExceededException.class) != null) {
                 // this can happen if for example a field is not_analyzed and ignore_above option is set.
                 // the field will be ignored when indexing but the huge term is still in the source and
                 // the plain highlighter will parse the source and try to analyze it.

--- a/src/test/java/org/elasticsearch/search/highlight/HighlighterSearchTests.java
+++ b/src/test/java/org/elasticsearch/search/highlight/HighlighterSearchTests.java
@@ -132,6 +132,10 @@ public class HighlighterSearchTests extends ElasticsearchIntegrationTest {
         search = client().prepareSearch().setQuery(constantScoreQuery(matchQuery("text", "text"))).addHighlightedField(new Field("long_text").highlighterType(highlighter)).get();
         assertNoFailures(search);
         assertThat(search.getHits().getAt(0).getHighlightFields().size(), equalTo(0));
+
+        search = client().prepareSearch().setQuery(prefixQuery("text", "te")).addHighlightedField(new Field("long_text").highlighterType(highlighter)).get();
+        assertNoFailures(search);
+        assertThat(search.getHits().getAt(0).getHighlightFields().size(), equalTo(0));
     }
 
     @Test

